File: java/src/main/java/ps_traning/programmers/level_1/푸드_파이터_대회.java
package ps_traning.programmers.level_1;

public class 푸드_파이터_대회 {
    public String solution(int[] food) {
        String answer = "";
        StringBuilder sb = new StringBuilder();
        for (int i = 1; i < food.length; i++) {
            for (int n = 0; n < (int) food[i] / 2; n++) {
                sb.append(i);
            }
        }
        String half = sb.toString();
        String reversed = sb.reverse().toString();
        sb.setLength(0);
        sb.append(half);
        sb.append(0);
        sb.append(reversed);
        answer = sb.toString();
        return answer;
    }
}

File: python/with_java/programmers/level 1/푸드 파이터 대회.py
def solution(food):
    half = []
    for i in range(1, len(food)):
        n = food[i] // 2
        for _ in range(n):
            half.append(i)
    reversed_half = list(reversed(half))
    answer = "".join(map(str, half + [0] + reversed_half))
    return answer